### PR TITLE
Better absolute path detection for Windows

### PR DESCRIPTION
**Pull request by mlohbihler**

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -25,8 +25,13 @@ var nodes = require('./nodes')
...
 
 exports.absolute = function(path){
-  // On Windows the path could start with a drive letter, i.e. a:\\ or two leading backslashes
-  return path.substr(0, 2) == '\\\\' || '/' === path.charAt(0) || /^[a-z]:\\/i.test(path);
+  // On Windows the path could start with a drive letter, i.e. a:\\ or two leading backslashes.
+  // Also on Windows, the drive letter can be capital, and the path may have been normalized to 
+  // forward slashes, so check for this too.
+  return path.substr(0, 2) == '\\\\'
+      || '/' === path.charAt(0)
+      || /^[a-zA-Z]:\\/i.test(path)
```
Comment (Panya) [inline lib/utils.js]: Last two checks could be simplified to `/^[a-z]:[\\\/]/i` (p.s. the `i` flag already means that this check is case-insensitive).

**Comment by Panya** [comment]
Could you provide a proof when absolute paths are normalized with forward slash on windows?  I can't find any info on that.

**Comment by mlohbihler** [comment]
Here is a sample of the paths which with the function is called. Note the variation in pathing symbols. I believe this began happening after Windows 10 was installed, but that was a while back and I can't be sure now so I can't say for certain whether this is a Windows thing or not. 

nib\index.styl
C:\Users\Matthew\git\stackhub\stackhub-ui\node_modules\stylus-loader\node_modules\stylus\lib\functions.styl
C:\Users\Matthew\git\stackhub\stackhub-ui\node_modules\stylus-loader\node_modules\stylus\lib\functions\index.styl
C:/Users/Matthew/git/stackhub/stackhub-ui/css/includes/breakpoints.styl
C:/Users/Matthew/git/stackhub/stackhub-ui/css/includes/layout.styl
C:/Users/Matthew/git/stackhub/stackhub-ui/css/includes/variables.styl

File: lib/utils.js (hunk 1 above)
Comment (mlohbihler) [inline lib/utils.js]: Yes, that code works.

**Comment by Panya** [comment]
Hmm, I haven't tested it with Windows 10 yet, but seems like some lib replaces `\` to `/` in these paths (I guess it's stylus-loader or something related to webpack).

Could you test your project with Stylus CLI compilation?

**Comment by mlohbihler** [comment]
At the moment i'm not set up to do so. (I'm really a server-side guy whose inherited this UI project.) The project does use stylus-loader, but there have been no updates to that recently. Does it matter if the fix is done there vs here?

**Comment by Panya** [comment]
>  Does it matter if the fix is done there vs here?

I just want to be sure that it is our bug. And I don't really want to fix someone's lib incorrect behaviour in our code base.

**Comment by mlohbihler** [comment]
Totally understandable. Still, it seems a pretty harmless change?

**Comment by savelichalex** [comment]
@mlohbihler Sorry, but I do same PR, I don't see that anyone do it too

**Comment by savelichalex** [comment]
@Panya I agree with you then maybe this lib use stylus code base incorrect, but in other hand this paths is correct in windows system

**Comment by Panya** [comment]
Ok-ok, merged :)

**Comment by mlohbihler** [comment]
:) Thanks Mikhail

**Pull request merged**